test/e2e/basic_test.go

```diff
@@ -338,19 +319,11 @@ func TestIPAMRestart(t *testing.T) {
 	}
 }
 
-// TestReconcileGatewayRoutesOnStartup checks that when the Antrea agent is restarted, the set of
+// testReconcileGatewayRoutesDuringStartup checks that when the Antrea agent is restarted, the set of
 // gateway routes is updated correctly, i.e. stale routes (for Nodes which are no longer in the
 // cluster) are removed and missing routes are added.
-func TestReconcileGatewayRoutesOnStartup(t *testing.T) {
+func testReconcileGatewayRoutesDuringStartup(t *testing.T, data *TestData) {
```

Comment:
This is actually mode relevant and it had different validation for different modes.

Reply:
revert it back to independent test case as `TestReconcileGatewayRoutesOnStartup`